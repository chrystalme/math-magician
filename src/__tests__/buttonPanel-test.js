import React from 'react';
import { render } from '@testing-library/react';
import ButtonPanel from '../components/buttonPanel';

describe('ButtonPanel', () => {
  const clickHandler = () => { ''; };
  test('ButtonPanel snapshot test', () => {
    const component = render(
      <ButtonPanel clickHandler={clickHandler} />,
    );
    expect(component).toMatchSnapshot();
  });
});
