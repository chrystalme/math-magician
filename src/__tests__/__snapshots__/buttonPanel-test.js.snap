// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel ButtonPanel snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex justify-content-around w-25 align-items-stretch"
      >
        <input
          class="m-2 p-3 btn btn-info false false"
          type="button"
          value="AC"
        />
        <input
          class="m-2 p-3 btn btn-info false false"
          type="button"
          value="+/-"
        />
        <input
          class="m-2 p-3 btn btn-info false false"
          type="button"
          value="%"
        />
        <input
          class="m-2 p-3 btn btn-danger false false"
          type="button"
          value="รท"
        />
      </div>
      <div
        class="d-flex justify-content-around w-25 align-items-stretch"
      >
        <input
          class="m-2 p-3 btn btn-secondary false false"
          type="button"
          value="7"
        />
        <input
          class="m-2 p-3 btn btn-secondary false false"
          type="button"
          value="8"
        />
        <input
          class="m-2 p-3 btn btn-secondary false false"
          type="button"
          value="9"
        />
        <input
          class="m-2 p-3 btn btn-danger false false"
          type="button"
          value="x"
        />
      </div>
      <div
        class="d-flex justify-content-around w-25 align-items-stretch"
      >
        <input
          class="m-2 p-3 btn btn-secondary false false"
          type="button"
          value="4"
        />
        <input
          class="m-2 p-3 btn btn-secondary false false"
          type="button"
          value="5"
        />
        <input
          class="m-2 p-3 btn btn-secondary false false"
          type="button"
          value="6"
        />
        <input
          class="m-2 p-3 btn btn-danger false false"
          type="button"
          value="-"
        />
      </div>
      <div
        class="d-flex justify-content-around w-25 align-items-stretch"
      >
        <input
          class="m-2 p-3 btn btn-secondary false false"
          type="button"
          value="1"
        />
        <input
          class="m-2 p-3 btn btn-secondary false false"
          type="button"
          value="2"
        />
        <input
          class="m-2 p-3 btn btn-secondary false false"
          type="button"
          value="3"
        />
        <input
          class="m-2 p-3 btn btn-danger false false"
          type="button"
          value="+"
        />
      </div>
      <div
        class="d-flex justify-content-around w-25 align-items-stretch"
      >
        <input
          class="m-2 p-3 btn btn-secondary w-50 false false"
          type="button"
          value="0"
        />
        <input
          class="m-2 p-3 btn btn-info false false"
          type="button"
          value="."
        />
        <input
          class="m-2 p-3 btn btn-info false false"
          type="button"
          value="="
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex justify-content-around w-25 align-items-stretch"
    >
      <input
        class="m-2 p-3 btn btn-info false false"
        type="button"
        value="AC"
      />
      <input
        class="m-2 p-3 btn btn-info false false"
        type="button"
        value="+/-"
      />
      <input
        class="m-2 p-3 btn btn-info false false"
        type="button"
        value="%"
      />
      <input
        class="m-2 p-3 btn btn-danger false false"
        type="button"
        value="รท"
      />
    </div>
    <div
      class="d-flex justify-content-around w-25 align-items-stretch"
    >
      <input
        class="m-2 p-3 btn btn-secondary false false"
        type="button"
        value="7"
      />
      <input
        class="m-2 p-3 btn btn-secondary false false"
        type="button"
        value="8"
      />
      <input
        class="m-2 p-3 btn btn-secondary false false"
        type="button"
        value="9"
      />
      <input
        class="m-2 p-3 btn btn-danger false false"
        type="button"
        value="x"
      />
    </div>
    <div
      class="d-flex justify-content-around w-25 align-items-stretch"
    >
      <input
        class="m-2 p-3 btn btn-secondary false false"
        type="button"
        value="4"
      />
      <input
        class="m-2 p-3 btn btn-secondary false false"
        type="button"
        value="5"
      />
      <input
        class="m-2 p-3 btn btn-secondary false false"
        type="button"
        value="6"
      />
      <input
        class="m-2 p-3 btn btn-danger false false"
        type="button"
        value="-"
      />
    </div>
    <div
      class="d-flex justify-content-around w-25 align-items-stretch"
    >
      <input
        class="m-2 p-3 btn btn-secondary false false"
        type="button"
        value="1"
      />
      <input
        class="m-2 p-3 btn btn-secondary false false"
        type="button"
        value="2"
      />
      <input
        class="m-2 p-3 btn btn-secondary false false"
        type="button"
        value="3"
      />
      <input
        class="m-2 p-3 btn btn-danger false false"
        type="button"
        value="+"
      />
    </div>
    <div
      class="d-flex justify-content-around w-25 align-items-stretch"
    >
      <input
        class="m-2 p-3 btn btn-secondary w-50 false false"
        type="button"
        value="0"
      />
      <input
        class="m-2 p-3 btn btn-info false false"
        type="button"
        value="."
      />
      <input
        class="m-2 p-3 btn btn-info false false"
        type="button"
        value="="
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
